interp/interpreter.go

```diff
@@ -427,6 +427,10 @@ func (r *runner) run(fn *function, params []value, parentMem *memoryView, indent
 				if err != nil {
 					return nil, mem, r.errorAt(inst, err)
 				}
+				if typecodePtr.offset() == 0 {
+					locals[inst.localIndex] = literalValue{uint8(0)}
+					break
+				}
```

Comment:
Can you please explain in a comment why this is correct? Because it is very non-obvious just reading the source code. It would need to include something about what `.offset() == 0` means (no method set), and why this implies the type assert is always false. Also, it's worth pointing out that `literalValue{uint8(0)}` is in fact setting the type assert value to false.

Reply:
I'm not sure commenting on that single piece of code is too useful, unless we comment also the logic after that. Before reading the writer (which is in another file) I couldn't understand any part of that logic. For example:

```go
typecodePtrOffset, err := typecodePtr.addOffset(-int64(r.pointerSize))
```

Why should we go back a pointer to get the typecode?
The reason is: because in [interface.go](https://github.com/tinygo-org/tinygo/blob/release/compiler/interface.go#L449), the typecodePtr always returns a pointer to the `numMethod` field. So, when the first member of typecode is the `methodsSet`, `typecodePtr` will point to the *second* member (offset != 0) and needs you must go back of a pointer size to get the method set. When it's not present, instead, it will point at the beginning (offset == 0) and the subtraction will fail (crashing the compiler).

So, maybe just a pointer to the method would be enough for a comment? Something along the lines of 

```go
// see compilerContext.getTypeCode in interface.go
```
could be enough?

Reply:
> Before reading the writer (which is in another file) I couldn't understand any part of that logic.

I admit I didn't explain the code well enough in comments when I wrote it. That doesn't mean we should keep the code hard-to-read. It's difficult to write too many comments (unless they're wrong).

Also, I've found that writing comments helps me avoid bugs. I've often found that once I tried to explain my code in a comment, I discovered my thinking was flawed and I had just written buggy code.

If you don't know what to write, you could write something like this:

> The offset is 0, which means the concrete type doesn't include a method set (the compiler doesn't generate method sets for concrete types without methods). The compiler doesn't emit interface type asserts for interfaces with no methods (they would always succeed) so this interface type assert must fail.

Reply:
Ok, I added a comment. I hope it's not too long/pedantic and it's clear. If not, just tell me and I'll try to reword it.